### DISCUSSION TITLE
After saving the many-to-many record, the browser field fails to display the name/title of the record

### DISCUSSION THREAD

**Discussion by Nvironmental** [area17/twill#1013] [Q&A]
Hi there, 

I am trying to implement a many-to-many relationship between my `MenuGroup` and `MenuItem` modules. So far I have added the following code in my `MenuItem` model 

``` 
public function menuGroups() {
       return $this->belongsToMany('App\Models\MenuGroup','menu_group_menu_item','menu_group_id','menu_item_id');
    }
```

in `MenuGroup` model

```
    public function menuItems () {
        return $this->belongsToMany('App\Models\MenuItem','menu_group_menu_item','menu_group_id','menu_item_id');
    }
```

in `MenuItemRepository`

```
 // implement the afterSave method
    public function afterSave($object, $fields) {
        $this->updateBrowser($object, $fields, 'menuGroups');
        parent::afterSave($object, $fields);
    }

    public function getFormFields($object)
    {
        $fields = parent::getFormFields($object);
        $fields['browsers']['menuGroups'] = $this->getFormFieldsForBrowser($object, 'menuGroups');
        return $fields;
    }

```

in `MenuGroupRepository`

```
       public function afterSave($object, $fields) {
        $this->updateBrowser($object, $fields, 'menuItems');
        parent::afterSave($object, $fields);
    }

    public function getFormFields($object)
    {
        $fields = parent::getFormFields($object);
        $fields['browsers']['menuItems'] = $this->getFormFieldsForBrowser($object, 'menuItems');
        return $fields;
    }
```

Menu Item Admin Blade File 
```
     @formField('browser', [
        'moduleName' => 'menuGroups',
        'name' => 'menuGroups',
        'label' => 'Menu Groups',
        'max' => 10
    ])
```

Menu Group Admin Blade File 
```
    @formField('browser', [
        'moduleName' => 'menuItems',
        'name' => 'menuItems',
        'label' => 'Menu Items',
        'max' => 10
    ])
```

However, I am getting the following output where the it fails to display the name/title of the records

in `menuItem`
![Screenshot 2020-03-31 at 9 38 46 PM](https://user-images.githubusercontent.com/31243747/78049462-b6f5b280-7398-11ea-91bb-10f79b488780.png)

in `menuGroup`
![Screenshot 2020-03-31 at 9 38 38 PM](https://user-images.githubusercontent.com/31243747/78049480-bbba6680-7398-11ea-942c-e434e35a0d28.png)

Is it a bug ? Please guide me as to what configuration/documentation I am missing.

Thank You. 🙏🏻

**Comment by galingong** [comment]
The function is known to work, this question should be asked on [Spectrum](https://spectrum.chat/twill), not on github. While your code looks fine (at first glance), from `2.0` on, you can use [automated browsers and repeaters](https://github.com/area17/twill/pull/400), so you should definitely do that, makes your life easier.

**Comment by ifox** [comment]
Does `MenuItem` have a `title` column? If not, you can define a `getTitleInBrowserAttribute` accessor.

**Comment by Nvironmental** [comment]
`getTitleInBrowserAttribute ` does the trick.  Tried [automated browsers and repeaters](https://github.com/area17/twill/pull/400) but had no luck. A bit more documentation will help.

**Comment by weaversam8** [comment]
Just stumbled into this issue myself. For reference of anyone facing this in the future, you implement the `getTitleInBrowserAttribute()` accessor on the model class.

@ifox - wouldn't it be wise to have this default use the `$titleColumnKey` property of the Module Controller?
